source/extensions/tracers/xray/tracer.cc

```diff
@@ -0,0 +1,124 @@
+#include "extensions/tracers/xray/tracer.h"
+
+#include <chrono>
+#include <ctime>
+#include <random>
+
+#include "common/common/utility.h"
+#include "common/tracing/http_tracer_impl.h"
+
+#include "extensions/tracers/xray/util.h"
+#include "extensions/tracers/xray/xray_core_constants.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+const std::string Tracer::VERSION_ = "1";
+const std::string Tracer::DELIMITER_ = "-";
+
+std::string Tracer::generateRandom96BitString() {
+  char values[25] = {'\0'};
+  auto seed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
```

Comment:
Use TestRandomGenerator for this, defined in test/test_common/utility.h which (a) centralizes this concept and (b) allows you to specify seeds from the test command line if you want to debug with the same behavior on each run.

Reply:
I don't want to call a test function in our source code. Instead, I use time_source_.systemTime() to be the seed.